scipy/signal/tests/test_signaltools.py

```diff
@@ -33,70 +32,91 @@
-        a = [3, 4, 5, 6, 5, 4]
-        b = [1, 2, 3]
+
+class TestConvolve:
+
+    @skip_xp_backends("jax.numpy", reason="dtypes do not match")
+    @array_api_compatible
+    def test_basic(self, xp):
+        a = xp.array([3, 4, 5, 6, 5, 4])
+        b = xp.array([1, 2, 3])
         c = convolve(a, b)
-        assert_array_equal(c, array([3, 10, 22, 28, 32, 32, 23, 12]))
+        xp_assert_equal(c, xp.array([3, 10, 22, 28, 32, 32, 23, 12]))
 
-    def test_same(self):
-        a = [3, 4, 5]
-        b = [1, 2, 3, 4]
+    @skip_xp_backends("jax.numpy", reason="dtypes do not match")
+    @array_api_compatible
+    def test_same(self, xp):
+        a = xp.array([3, 4, 5])
+        b = xp.array([1, 2, 3, 4])
         c = convolve(a, b, mode="same")
-        assert_array_equal(c, array([10, 22, 34]))
+        xp_assert_equal(c, xp.array([10, 22, 34]))
 
-    def test_same_eq(self):
-        a = [3, 4, 5]
-        b = [1, 2, 3]
+    @skip_xp_backends("jax.numpy", reason="dtypes do not match")
+    @array_api_compatible
+    def test_same_eq(self, xp):
+        a = xp.array([3, 4, 5])
+        b = xp.array([1, 2, 3])
         c = convolve(a, b, mode="same")
-        assert_array_equal(c, array([10, 22, 22]))
+        xp_assert_equal(c, xp.array([10, 22, 22]))
 
-    def test_complex(self):
-        x = array([1 + 1j, 2 + 1j, 3 + 1j])
-        y = array([1 + 1j, 2 + 1j])
+    @array_api_compatible
+    def test_complex(self, xp):
+        x = xp.array([1 + 1j, 2 + 1j, 3 + 1j])
+        y = xp.array([1 + 1j, 2 + 1j])
         z = convolve(x, y)
-        assert_array_equal(z, array([2j, 2 + 6j, 5 + 8j, 5 + 5j]))
+        xp_assert_equal(z, xp.array([2j, 2 + 6j, 5 + 8j, 5 + 5j]))
 
-    def test_zero_rank(self):
+    @array_api_compatible
+    def test_zero_rank(self, xp):
         a = 1289
         b = 4567
         c = convolve(a, b)
-        assert_equal(c, a * b)
+        xp_assert_equal(c, a * b)
 
-    def test_broadcastable(self):
-        a = np.arange(27).reshape(3, 3, 3)
-        b = np.arange(3)
+    @array_api_compatible
+    def test_broadcastable(self, xp):
+        a = xp.arange(27).reshape(3, 3, 3)
+        b = xp.arange(3)
         for i in range(3):
             b_shape = [1]*3
             b_shape[i] = 3
             x = convolve(a, b.reshape(b_shape), method='direct')
             y = convolve(a, b.reshape(b_shape), method='fft')
-            assert_allclose(x, y)
+            xp_assert_close(x, y, atol=1e-14)
 
-    def test_single_element(self):
-        a = array([4967])
-        b = array([3920])
+    @array_api_compatible
+    def test_single_element(self, xp):
+        a = np.array([4967])
+        b = np.array([3920])
         c = convolve(a, b)
-        assert_equal(c, a * b)
-
-    def test_2d_arrays(self):
-        a = [[1, 2, 3], [3, 4, 5]]
-        b = [[2, 3, 4], [4, 5, 6]]
+        xp_assert_equal(c, a * b)
+
+ #   @skip_xp_backends("jax.numpy")   XXX: how to skip two backends
+    @skip_xp_backends("cupy")
```

Comment:
```suggestion
    @skip_xp_backends("jax.numpy", "cupy")
```

Reply:
refer to the docstring at https://github.com/scipy/scipy/blob/b421cd64d98c16811f84efbfab7701b335f811be/scipy/conftest.py#L173-L199.

The parameter is called `reasons` not `reason`, and accepts a `list(str)`